src/components/Drawer/__tests__/Drawer.spec.js

```diff
@@ -2,50 +2,107 @@ import React from 'react';
         );
 
-        expect(wrapper.find('.Drawer').hasClass('Drawer--isOpen')).toBe(true);
-        expect(wrapper.find('.Drawer__extendButton').hasClass('Drawer__extendButton--isOpen')).toBe(
-            true
-        );
+        expect(wrapper.find('.Drawer').hasClass('Drawer--isShownAndExpanded')).toBe(true);
+        expect(wrapper.find('.Drawer').hasClass('Drawer--isShownAndClosed')).toBe(false);
+        expect(
+            wrapper.find('.Drawer__expandButton').hasClass('Drawer__expandButton--isExpanded')
+        ).toBe(true);
     });
 
     it('should pass initial close status', () => {
-        const wrapper = shallow(<Drawer title="some title">some text</Drawer>);
+        const wrapper = mount(<Drawer title="some title">some text</Drawer>);
+
+        expect(wrapper.find('.Drawer').hasClass('Drawer--isShownAndClosed')).toBe(true);
+        expect(wrapper.find('.Drawer').hasClass('Drawer--isShownAndExpanded')).toBe(false);
+        expect(
+            wrapper.find('.Drawer__expandButton').hasClass('Drawer__expandButton--isExpanded')
+        ).toBe(false);
+    });
+
+    it('should expand and close correctly', () => {
+        const wrapper = mount(<Drawer title="some title">some text</Drawer>);
+        // Expand Drawer
+        wrapper.find('.Drawer__expandButton').simulate('click');
+
+        expect(wrapper.find('.Drawer').hasClass('Drawer--isShownAndExpanded')).toBe(true);
+        expect(wrapper.find('.Drawer').hasClass('Drawer--isShownAndClosed')).toBe(false);
+        expect(
+            wrapper.find('.Drawer__expandButton').hasClass('Drawer__expandButton--isExpanded')
+        ).toBe(true);
 
-        expect(wrapper.find('.Drawer').hasClass('Drawer--isOpen')).toBe(false);
-        expect(wrapper.find('.Drawer__extendButton').hasClass('Drawer__extendButton--isOpen')).toBe(
-            false
+        // Close Drawer
+        wrapper.find('.Drawer__expandButton').simulate('click');
+
+        expect(wrapper.find('.Drawer').hasClass('Drawer--isShownAndClosed')).toBe(true);
+        expect(wrapper.find('.Drawer').hasClass('Drawer--isShownAndExpanded')).toBe(false);
+        expect(
+            wrapper.find('.Drawer__expandButton').hasClass('Drawer__expandButton--isExpanded')
+        ).toBe(false);
+    });
+
+    it('should be hidden when isShown is false', () => {
+        const wrapper = mount(
+            <Drawer isShown={false} title="some title">
+                some text
+            </Drawer>
         );
+
+        expect(wrapper.find('.Drawer').hasClass('Drawer--isShownAndExpanded')).toBe(false);
+        expect(wrapper.find('.Drawer').hasClass('Drawer--isShownAndClosed')).toBe(false);
     });
 
-    it('should open and close correctly', () => {
-        const wrapper = shallow(<Drawer title="some title">some text</Drawer>);
-        // Open Drawer
-        wrapper.find('.Drawer__extendButton').simulate('click');
+    it('should be still hidden when drawer expented and isShown is false', () => {
```

Comment:
typo: expented => expanded

Reply:
expanted => expanded ;-)

I have this spell checker extension in VSCode now. I'm usually terrible with this too, but it helps a lot. Checkout: Code Spell Checker
streetsidesoftware.code-spell-checker. It manages all the CamerCase words too.

Reply:
Awesome! I will try it. Thanks!